websocket_js.go

```diff
@@ -89,16 +114,30 @@ func (c *Conn) Read(ctx context.Context) (MessageType, []byte, error) {
 }
 
 func (c *Conn) read(ctx context.Context) (MessageType, []byte, error) {
-	var me wsjs.MessageEvent
 	select {
 	case <-ctx.Done():
 		c.Close(StatusPolicyViolation, "read timed out")
 		return 0, nil, ctx.Err()
-	case me = <-c.readch:
+	case <-c.readSignal:
 	case <-c.closed:
 		return 0, nil, c.closeErr
 	}
 
+	c.readBufMu.Lock()
+	defer c.readBufMu.Unlock()
+
+	me := c.readBuf[0]
+	copy(c.readBuf, c.readBuf[1:])
```

Comment:
Hmm.. this version requires fewer memory allocations but it also calls `copy` on every read. I'm curious if this has any impact on performance? It might be fine.

Reply:
I don't expect it to be a problem. Copying data is very fast, especially since we're only copying pointers to the data.